crates/rust-analyzer/src/main_loop.rs

```diff
@@ -549,8 +549,11 @@ impl GlobalState {
                 let subscriptions = subscriptions.clone();
                 // Do not fetch semantic diagnostics (and populate query results) if we haven't even
                 // loaded the initial workspace yet.
-                let fetch_semantic =
-                    self.vfs_done && self.fetch_workspaces_queue.last_op_result().is_some();
+                let fetch_semantic = self.vfs_done
+                    && self
+                        .fetch_workspaces_queue
+                        .last_op_result()
+                        .is_some_and(|inner| inner.is_some());
```

Comment:
We should be able to get rid of the `Option` inside of the `fetch_workspaces_queue` field now. It serves the same purpose as your changes (discerning between initial run) so the double nesting of options is pointless

Reply:
Done! I wanted to be conservative with my changes, but I removed the nested option.

(I also switched the "response" of `fetch_workspaces_queue` to a struct from an tuple. I'll switch the remaining OpQueues to use named structs instead of tuples in a subsequent PR.)